### ISSUE TITLE
Dummy records inserted into Kafka topics

### ISSUE THREAD

**Issue by GeorgiBonchev** [zendesk/maxwell#1978]
Hello,

We have the following problem. Suddenly Maxwell started to insert some "Dummy" records into several different Kafka topics. The records look like this:

{
  "commit": true,
  "data": {},
  "database": "maxwell",
  "push_ts": 1.676387195779338e+09,
  "table": "dummy",
  "ts": 1676387195,
  "type": "insert"
}

Only a few records are inserted randomly as it seems but they are enough to break our pipeline.

We are using version 1.37.6 of Maxwell

Any idea what might be the problem?

**Comment by osheroff** [comment]
What version of MySQL?  I assume Maria db?On Feb 15, 2023, at 10:28, GEORGI BONCHEV ***@***.***> wrote:﻿
Hello,
We have the following problem. Suddenly Maxwell started to insert some "Dummy" records into several different Kafka topics. The records look like this:
{
"commit": true,
"data": {},
"database": "maxwell",
"push_ts": 1.676387195779338e+09,
"table": "dummy",
"ts": 1676387195,
"type": "insert"
}
Only a few records are inserted randomly as it seems but they are enough to break our pipeline.
We are using version 1.37.6 of Maxwell
Any idea what might be the problem?

**Comment by GeorgiBonchev** [comment]
> What version of MySQL?  I assume Maria db?

mysql Ver 14.14 Distrib 5.7.20-18, for Linux (x86_64) using 6.0

**Comment by nsveli** [comment]
I have the same issue, can you follow up, please? Does the dummy record come from the getLatency method? When does the getLatency() method gets called and why is there a dummy insert logic inside the function?
```
	public CompletableFuture<Long> getLatency() {
		DiagnosticCallback callback = new DiagnosticCallback();
		try {
			RowMap rowMap = new RowMap("insert", config.databaseName, "dummy", System.currentTimeMillis(),
					new ArrayList<>(), positionStoreThread.getPosition());
			rowMap.setTXCommit();
			ProducerRecord<String, String> record = producer.makeProducerRecord(rowMap);
			producer.sendAsync(record, callback);
		} catch (Exception e) {
			callback.latency.completeExceptionally(e);
		}
		return callback.latency;
	}
```

**Comment by wanghangyu817** [comment]
@GeorgiBonchev Do you have `http_diagnostic=true` configured?

**Comment by osheroff** [comment]
so this code is definitely the source of these records, but it's strange that they made it into the kafka topic; records inserted for the `maxwell` database (aka `config.databaseName`) should be explicitly filtered out in a separate part of the code: https://github.com/zendesk/maxwell/blob/b13e21c9281b7911c5ede454c9d7c3e0e4c345b3/src/main/java/com/zendesk/maxwell/replication/BinlogConnectorReplicator.java#L348

**Comment by osheroff** [comment]
https://github.com/zendesk/maxwell/blob/b13e21c9281b7911c5ede454c9d7c3e0e4c345b3/src/main/java/com/zendesk/maxwell/replication/BinlogConnectorReplicator.java#L323-L326

**Comment by GeorgiBonchev** [comment]
@wanghangyu817 Yes, I just checked configuration. http_diagnostic is set to true. Should I set it to false?

**Comment by wanghangyu817** [comment]
Yes, if you don't want to see this data, set `http_diagnostic` to `false`

**Comment by wanghangyu817** [comment]
https://github.com/zendesk/maxwell/blob/b13e21c9281b7911c5ede454c9d7c3e0e4c345b3/src/main/java/com/zendesk/maxwell/monitoring/DiagnosticHealthCheck.java#L44-L45
https://github.com/zendesk/maxwell/blob/b13e21c9281b7911c5ede454c9d7c3e0e4c345b3/src/main/java/com/zendesk/maxwell/producer/KafkaProducerDiagnostic.java#L52-L56

`sendAsync()` method is called directly after `RowMap` is built, and data is pushed directly to kafka. `shouldSkipRow()` does not work in this process;Maybe there's something we can add to the data to make sure the format is consistent and not empty

**Comment by osheroff** [comment]
oh right.  derp.

**Comment by osheroff** [comment]
ok I suppose this is working as intended, although it definitely violates principle-of-least-surprise.

**Comment by nsveli** [comment]
thank you very much for your cooperation!
